pkg/component/server/etcd.go

```diff
@@ -66,28 +59,45 @@ func (e *Etcd) Init() error {
 		logrus.Warning(errors.Wrap(err, "Running etcd as root"))
 	}
 
-	err = util.InitDirectory(constant.EtcdDataDir, constant.EtcdDataDirMode) // https://docs.datadoghq.com/security_monitoring/default_rules/cis-kubernetes-1.5.1-1.1.11/
+	err = util.InitDirectory(e.K0sVars.EtcdDataDir, constant.EtcdDataDirMode) // https://docs.datadoghq.com/security_monitoring/default_rules/cis-kubernetes-1.5.1-1.1.11/
 	if err != nil {
-		return errors.Wrapf(err, "failed to create %s", constant.EtcdDataDir)
+		return errors.Wrapf(err, "failed to create %s", e.K0sVars.EtcdDataDir)
 	}
 
-	err = util.InitDirectory(constant.EtcdCertDir, constant.EtcdCertDirMode) // https://docs.datadoghq.com/security_monitoring/default_rules/cis-kubernetes-1.5.1-4.1.7/
+	err = util.InitDirectory(e.K0sVars.EtcdCertDir, constant.EtcdCertDirMode) // https://docs.datadoghq.com/security_monitoring/default_rules/cis-kubernetes-1.5.1-4.1.7/
 	if err != nil {
 		return errors.Wrapf(err, "failed to create etcd cert dir")
 	}
 
-	for _, f := range []string{constant.EtcdDataDir, constant.EtcdCertDir} {
+	for _, f := range []string{e.K0sVars.EtcdDataDir, e.K0sVars.EtcdCertDir} {
 		err = os.Chown(f, e.uid, e.gid)
 		if err != nil && os.Geteuid() == 0 {
 			return err
 		}
 	}
 
-	return assets.Stage(constant.BinDir, "etcd", constant.BinDirMode)
+	for _, f := range []string{
+		"ca.crt",
+		"server.crt",
+		"server.key",
+	} {
+		if err := os.Chown(path.Join(e.K0sVars.EtcdCertDir, f), e.uid, e.gid); err != nil && os.Geteuid() == 0 {
+			// TODO: directory may not yet exist. log it and wait for retry for now
+			logrus.Errorf("failed to chown %s: %s", f, err)
+		}
+	}
+	return assets.Stage(e.K0sVars.BinDir, "etcd", constant.BinDirMode)
```

Comment:
I don't think we should re-introduce this. See commit 3daba31160eb1c83ae4abf6c0e6a6553a6ba5182